pkg/operator/serviceaccountissuercontroller/serviceaccountissuer_controller_test.go

```diff
@@ -83,22 +83,30 @@ func TestController(t *testing.T) {
 					Name: "cluster",
 				},
 			},
-			expectedStatus: []operatorv1.ServiceAccountIssuerStatus{},
+			expectedResync: true,
+			expectedStatus: defaultServiceAccountIssuerValue,
 		},
 		{
-			name: "serviceaccountissuer is set in auth config and should be copied to status",
+			name: "serviceaccountissuer is set in auth config and should be copied to status while making default issuer trusted",
 			authConfig: configv1.Authentication{Spec: configv1.AuthenticationSpec{
 				ServiceAccountIssuer: "newIssuer",
 			}},
 			operator: operatorv1.KubeAPIServer{
 				ObjectMeta: metav1.ObjectMeta{
 					Name: "cluster",
 				},
+				Status: operatorv1.KubeAPIServerStatus{
+					ServiceAccountIssuers: defaultServiceAccountIssuerValue,
+				},
 			},
```

Comment:
Add a test for when you configure the issuer specifically to `defaultServiceAccountIssuerValue[0].Name` and previous issuers are set.

The result should be that previous issuer is still trusted and the default issuer is the current one - compared to when you would set the issuer to an empty string.

Reply:
test added